R/sparklyr-spark-connect.R

```diff
@@ -71,22 +72,28 @@ spark_connect_method.spark_method_databricks_connect <- function(
   method <- method[[1]]
   token <- databricks_token(token, fail = FALSE)
```

Comment:
Based on your comment on line 137, I think we should remove this line. And have `token` only populated when the user passes it as an argument in the `spark_connect()` call

Reply:
My thinking for leaving this was that users explicitly setting the `DATABRICKS_TOKEN` and `DATABRICKS_HOST` vars should have those respected as it were set explicitly. The Databricks Python SDK won't detect those when its done from R.

`databricks_token` function also looks for `CONNECT_DATABRICKS_TOKEN` so i think its probably important to leave that intact?

I was expecting hierarchy to be:
1. Explicit `token`
2. `DATABRICKS_TOKEN`
3. `CONNECT_DATABRICKS_TOKEN`
4. `.rs.api.getDatabricksToken(host)`
5. Python SDK explicit setting of profile
6. Python SDK detection of `DEFAULT` profile